frontend/src/components/AuthForm.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="auth-form">
+    <h2>{{ isLogin ? 'Login' : 'Register' }}</h2>
+    <form @submit.prevent="handleSubmit">
+      <input 
+        v-if="!isLogin" 
+        v-model="username" 
+        placeholder="Username" 
+        required 
+      />
+      <input 
+        v-model="email" 
+        type="email" 
+        placeholder="Email" 
+        required 
+      />
+      <input 
+        v-model="password" 
+        type="password" 
+        placeholder="Password" 
+        required 
+      />
+      <button type="submit">{{ isLogin ? 'Login' : 'Register' }}</button>
+    </form>
+    <p v-if="message">{{ message }}</p>
+    <button @click="toggleMode">
+      <router-link :to="isLogin ? '/register' : '/login'">
```

Comment:
Nisam siguran šta bi trebalo da se desi u ovom slučaju i kojim redosledom će da se desi. :) Praktično imaš `<a>` u okviru `<button>` što je malo neuobičajeno. Neki best practice je da se button koristi za stvari koje se dešavaju na stranici, znači neka interna funkcionalnost, a <a> ako se linkuje na neku drugu stranicu, bilo internu, bilo eksternu. U ovom slučaju linkuješ na `login` i `register` stranice, tako da je button verovatno višak. Pretpostavljam da je poenta da button promeni `isLogin` mode, ali to bi onda trebalo da rešiš na neki drugi način. Zapravo, mislim da možeš samo da premestiš click event na url sa dugmeta, izbaciš dugme, a url stilizuješ preko CSS da izgleda kao dugme.